samples/opengl/depthMapTriangulation/depthMapTriangulation.cpp

```diff
@@ -0,0 +1,313 @@
+    cameraParameters = StereoCamera4Base<float>(5.3887405952849110e+02, 5.3937051275591125e+02, 3.2233507920081263e+02,
+                                                2.3691517848391885e+02, 40.0f);
+    ip_settings.cameraParameters  = cameraParameters;
+    st_settings.cameraParameters  = cameraParameters;
+    rqt_settings.cameraParameters = cameraParameters;
+
+    // The necessary calls to get a naive triangulated mesh
+    load_depth_image();
+    scale_down_depth_image();
+    preprocess_occlusion_edges();
+    blur_depth_image();
+    triangulate_naive();
+
+
+    // This simple AssetLoader can create assets from meshes and generate some generic debug assets
+    ObjAssetLoader assetLoader;
+
+    meshObject.asset = assetLoader.assetFromMesh(depthmesh);
+    meshObject.translateGlobal(vec3(0, 1, 0));
+    meshObject.calculateModel();
+
+    std::cout << "Program Initialized!" << std::endl;
+}
+
+void Sample::load_depth_image()
+{
+    loaded_depth_image = TemplatedImage<float>(depth_image_input);
+
+    rqt_settings.image_height = loaded_depth_image.height;
+    rqt_settings.image_width  = loaded_depth_image.width;
+
+    // set the camera parameters of all settings to the values of the used sample image
+    cameraParameters = StereoCamera4Base<float>(5.3887405952849110e+02, 5.3937051275591125e+02, 3.2233507920081263e+02,
+                                                2.3691517848391885e+02, 40.0f);
+    ip_settings.cameraParameters  = cameraParameters;
+    st_settings.cameraParameters  = cameraParameters;
+    rqt_settings.cameraParameters = cameraParameters;
+}
+
+void Sample::scale_down_depth_image()
+{
+    DMPPParameters dmppp = DMPPParameters();
+
+    Intrinsics4 intrinsics(cameraParameters.fx, cameraParameters.fy, cameraParameters.cx, cameraParameters.cy);
+
+    DMPP dmpp(intrinsics, dmppp);
+
+    if (loaded_depth_image.height % 2 != 0 || loaded_depth_image.width % 2 != 0)
+    {
+        std::cout << "can't scale down image with odd dimension.\n";
+        return;
+    }
+
+    TemplatedImage<float> result(loaded_depth_image.height / 2, loaded_depth_image.width / 2);
+
+    // scale down the image
+    dmpp.scaleDown2median(loaded_depth_image, result);
+
+    // also scale down camera parameters
+    cameraParameters.scale(0.5f);
+    ip_settings.cameraParameters  = cameraParameters;
+    st_settings.cameraParameters  = cameraParameters;
+    rqt_settings.cameraParameters = cameraParameters;
+
+    rqt_settings.image_height /= 2;
+    rqt_settings.image_width /= 2;
+
+    loaded_depth_image = result;
+}
+
+void Sample::preprocess_occlusion_edges()
+{
+    ImageProcessor ip(ip_settings);
+
+    ip.remove_occlusion_edges(loaded_depth_image);
+}
+
+void Sample::blur_depth_image()
+{
+    TemplatedImage<float> result(loaded_depth_image.height, loaded_depth_image.width);
+
+    ImageProcessor ip(ip_settings);
+
+    ip.filter_gaussian(loaded_depth_image, result);
+
+    loaded_depth_image = result;
```

Comment:
remove unnecessary empty lines. Please check that on all files. I won't comment it everywhere.